### PR TITLE
Project polish

### PR DESCRIPTION
**Pull request by Developers-codz**

### CODE REVIEW

**Review by sharathnair9999 — APPROVED** [review]
Looks good. All the refactoring made it more sensible. Specially the user icon part 😃. Good work

File: src/component/navbar/navbar.jsx [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
- 
-  const changeHandler = () =>{
-      dispatch({type:"SEARCH",payload: searchText}) 
-  }
+  const changeHandler = () => {
+    dispatch({ type: "SEARCH", payload: searchText });
+  };
 
-  const getOptimisedVersion = debounce(changeHandler,1000);
+  const getOptimisedVersion = debounce(changeHandler, 1000);
```
Comment (vishalg8454) [inline src/component/navbar/navbar.jsx:25]: what about a better naming like `debouncedChangeHandler` or `optimizedChangeHandler` ?

**Review by vishalg8454 — APPROVED** [review]
Good to merge!

File: src/component/navbar/navbar.jsx (hunk 1 above)
Comment (Developers-codz) [reply src/component/navbar/navbar.jsx:25]: I think this name also makes sense but thanks for the suggestion 😁

**Pull request merged**